lib/core/src/Network/Ntp.hs

```diff
@@ -121,8 +123,11 @@ instance HasSeverityAnnotation NtpTrace where
         NtpTracePacketReceived _ _ -> Debug
         NtpTraceWaitingForRepliesTimeout _ -> Notice
 
-getNtpStatus :: NtpClient -> IO ApiNetworkClock
-getNtpStatus = fmap (ApiNetworkClock . toStatus) . ntpQueryBlocking
+getNtpStatus :: NtpClient -> Bool -> IO ApiNetworkClock
+getNtpStatus client forceCheck = (ApiNetworkClock . toStatus) <$>
+    if forceCheck
+    then ntpQueryBlocking client
+    else atomically (ntpGetStatus client)
```

Comment:
Perhaps a comment that ntpQueryBlocking does not use a cached result but ntpGetStatus does.
It's not really obvious from the names.

Reply:
Good point :+1:

Reply:
I've commented everywhere else but here :upside_down_face: .....

Reply:
done in https://github.com/input-output-hk/cardano-wallet/commit/3df212532ad60888d7a98e78285cba395356cc25

Reply:
@KtorZ I think you want to do this:
```
atomically $ do
  s <- ntpGetStatus client
  check (s /= NtpSyncPending)
  pure s
```
Which will block for at most `1s`, if the ntp is currently running.

Reply:
@rvl if it's blocking it surly must do something; We could rename `ntpGetStatus` as `ntpGetStatusNonBlocking` but since it returns an `STM` action it cannot run the ntp protocol (unless we are evil and run `IO` inside `STM` ;).

Reply:
@coot rather, `ntpGetCachedStatus` could better convey the idea IMO :+1: 

> (unless we are evil and run IO inside STM

Expect the unexpectable. We will stop at nothing.